### PR TITLE
add test for report fixture reference

### PR DESCRIPTION
**Pull request by czue**
@emord https://github.com/dimagi/commcare-hq/pull/13571#issuecomment-253818568

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_advanced_suite.py [hunk 1]
```diff
@@ -196,6 +196,23 @@ def test_advanced_suite_load_case_from_fixture_with_custom_fixture(self):
...
+        app.get_module(1).get_form(0).actions.load_update_cases.append(LoadUpdateAction(
+            case_tag="",
+            case_type="clinic",
+            load_case_from_fixture=LoadCaseFromFixture(
+                fixture_nodeset="instance('commcare:reports')/reports/report[@id='some-report-guid']/rows/row",
+                fixture_tag="selected_row",
+                fixture_variable="index",
+            )
+        ))
+        with patch('corehq.apps.app_manager.suite_xml.post_process.instances.toggles.MOBILE_UCR.enabled') as mock_toggle:
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: E501 line too long (121 > 115 characters)

File: corehq/apps/app_manager/tests/test_advanced_suite.py [hunk 2]
```diff
@@ -196,6 +196,23 @@ def test_advanced_suite_load_case_from_fixture_with_custom_fixture(self):
...
+            mock_toggle.return_value = True
+            suite = app.create_suite()
+        self.assertXmlPartialEqual(self.get_xml('load_case_from_report_fixture_session'), suite, './entry[2]/session')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/app_manager/tests/test_advanced_suite.py [hunk 3]
```diff
@@ -196,6 +196,23 @@ def test_advanced_suite_load_case_from_fixture_with_custom_fixture(self):
...
+        self.assertXmlPartialEqual(self.get_xml('load_case_from_report_fixture_instance'), suite, './entry[2]/instance')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: E501 line too long (120 > 115 characters)

**Review by emord — APPROVED** [review]

File: corehq/apps/app_manager/tests/test_advanced_suite.py (hunk 1 above)
Comment (proteusvacuum) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: I think you can use the [`flag_enabled`](https://github.com/dimagi/commcare-hq/blob/ed6846138b57ba8c370f01fe9cf72a613cf09ca3/corehq/util/test_utils.py#L119-L130) context manager

File: corehq/apps/app_manager/tests/test_advanced_suite.py [hunk 4]
```diff
@@ -190,12 +191,26 @@ def test_advanced_suite_load_case_from_fixture_with_custom_fixture(self):
...
+            case_tag="",
+            case_type="clinic",
+            load_case_from_fixture=LoadCaseFromFixture(
+                fixture_nodeset="instance('commcare:reports')/reports/report[@id='some-report-guid']/rows/row",
+                fixture_tag="selected_row",
+                fixture_variable="index",
+            )
+        ))
+        suite = app.create_suite()
+        self.assertXmlPartialEqual(self.get_xml('load_case_from_report_fixture_session'), suite, './entry[2]/session')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/app_manager/tests/test_advanced_suite.py [hunk 5]
```diff
@@ -190,12 +191,26 @@ def test_advanced_suite_load_case_from_fixture_with_custom_fixture(self):
...
+        self.assertXmlPartialEqual(self.get_xml('load_case_from_report_fixture_instance'), suite, './entry[2]/instance')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: E501 line too long (120 > 115 characters)

File: corehq/apps/app_manager/tests/test_advanced_suite.py (hunk 1 above)
Comment (czue) [inline corehq/apps/app_manager/tests/test_advanced_suite.py]: nice: https://github.com/dimagi/commcare-hq/pull/13577/commits/320f21d7896e487bde9bee95056891fd9cd5d371

**Pull request merged**